### DISCUSSION TITLE
Consulta sobre punto 2 TPA3

### DISCUSSION THREAD

**Discussion by agmblanco** [dds-jv/foro#440] [TPA - 2024] [answered]
Buenas noches profes. Tenemos una problema con el requerimiento número 2 de la tercer entrega: "_Se deben registrar tanto las solicitudes de apertura en las heladeras como las aperturas hechas por los colaboradores._". Con la primer parte no tenemos problema; pero con la segunda, no entendemos cómo haríamos eso, ya que a nuestro entender, el dispositivo colocado en la heladera física y con quien nuestro sistema tiene que interactuar para autorizar las tarjetas de los colaboradores a realizar una apertura no soporta un método que nos permita registrar un callback para el evento de apertura de heladera. No tendríamos forma de enterarnos si realmente el colaborador realizó o no la apertura solicitada. 
Podemos asumir que de esos registros se encarga el dispositivo de la heladera? 

Saludos!

**Comment by flbulgarelli** [comment, marked as answer]
> no soporta un método que nos permita registrar un callback para el evento de apertura de heladera. No tendríamos forma de enterarnos si realmente el colaborador realizó o no la apertura solicitada.

Esto es cierto, en parte. No hay un método que indique quien abrió la heladera, pero sí hay un evento para el cambio de cantidades de viandas. Aunque es rudimentario, eso sí deberían registrarlo como parte de ese historial o registro de actividad. 

> Podemos asumir que de esos registros se encarga el dispositivo de la heladera?

Para todo lo demás, sí, dado que no hay mucho más que se pueda hacer.

**Reply by agmblanco** [reply]
Perfecto, muchas gracias!!
